pkg/ephemeral/io/carrier.go

```diff
@@ -42,16 +46,47 @@ type Config struct {
 }
 
 // Connect establishes a TCP connection to a socket on a given host and port.
-func (c *Carrier) Connect(ctx context.Context, host, port string) error {
+func (c *Carrier) Connect(ctx context.Context, playerID int32, host string, port string) error {
 	conn, err := c.Dialer(ctx, host, port)
 	if err != nil {
 		return err
 	}
-	c.Conn = conn
+	_, err = conn.Write(c.buildHeader(playerID))
+	if err != nil {
+		return err
+	}
+	c.Conn, err = c.TLSConnector(conn, playerID)
+	if err != nil {
+		return err
+	}
+	if playerID == 0 {
+		err = c.readSpec()
+		if err != nil {
+			return err
+		}
+	}
 	c.connected = true
 	return nil
 }
 
+func (c Carrier) readSpec() error {
```

Comment:
Could you please add the description including the link to the MP-SPDZ repository comment you provided in the respective test case as an answer to @sbckr's question regarding the format?

Reply:
Done.